SourceX/controls/devices/game_controller.cpp

```diff
@@ -150,6 +153,19 @@ bool GameController::ProcessAxisMotion(const SDL_Event &event)
 
 void GameController::Add(int joystick_index)
 {
+#ifdef __vita__
```

Comment:
Let's make this non-Vita specific:

```c++
getIniValue("controller", "extra_sdl2_mapping", ...)
```

It also should happen at SDL initialization instead of here.

Alternatively, instead of doing this in code, we can:

a. Set `SDL_GAMECONTROLLERCONFIG` env var in the launcher.
b. Add this mapping to the SDL2 shipped with the Vita SDK.

Reply:
a. won't work, there's no env on Vita
b. There's default mapping in Vita SDL2, but this option is to allow re-mapping controls on Vita.

Making this non-Vita specific would require to include controller name/guid in ini setting. Though, that can be dealt with by mentioning this in readme.

Reply:
Yes please, let's make this non Vita-specific

The user will begin by pasting the whole string from the README anyway, so the user experience isn't impacted much

Reply:
Made this non-vita specific, moved to sdl initialisation and moved to `[controls]` ini section as `sdl2_controller_mapping`